sync-core/src/main/java/com/cloudant/sync/datastore/RevisionHistoryHelper.java

```diff
@@ -195,7 +195,8 @@ public static MultipartAttachmentWriter createMultipartWriter(BasicDocumentRevis
                         mpw = new MultipartAttachmentWriter();
                         mpw.setBody(revision);
                     }
-                    mpw.addAttachment(att);
+                    mpw.addAttachment(savedAtt,
+                            savedAtt.encoding == Attachment.Encoding.Plain ? savedAtt.length : savedAtt.encodedLength);
```

Comment:
We have similar logic to this in several places. Could we localise into methods like "dataLength" and "onDiskLength" somewhere?